Makefile.toml

```diff
@@ -547,21 +547,55 @@ pubsys \
 ]
 
 [tasks.clean]
+dependencies = [
+  "clean-sources",
+  "clean-packages",
+  "clean-images",
+  "clean-repos",
+]
+
+[tasks.clean-sources]
 script_runner = "bash"
 script = [
 '''
-for ws in sources packages variants/* tools/{buildsys,pubsys}; do
+for ws in sources variants/* tools/{buildsys,pubsys}; do
   [ -d "${ws}" ] || continue
   cargo clean --manifest-path ${ws}/Cargo.toml
 done
 rm -f ${BUILDSYS_TOOLS_DIR}/bin/{buildsys,pubsys}
-for ext in tar lz4 img ; do
-  rm -f ${BUILDSYS_OUTPUT_DIR}/*.${ext}
+'''
+]
+
+[tasks.clean-packages]
+script_runner = "bash"
+script = [
+'''
+for ws in packages; do
+  [ -d "${ws}" ] || continue
+  cargo clean --manifest-path ${ws}/Cargo.toml
 done
+rm -rf ${BUILDSYS_PACKAGES_DIR}
+'''
+]
+
+# Note: this only cleans images for the current arch and variant.  Specify
+# BUILDSYS_ARCH and/or BUILDSYS_VARIANT in your call to clean images for other
+# combinations.
+[tasks.clean-images]
```

Comment:
Why this restriction? All of the other ones clean up all their respective content.

I switch between variants and arches quite a lot, and I'd prefer to have an easy way to clean up all the images.

Reply:
I was just documenting the current restriction for now. I agree that this is a little confusing, but I thought it'd be a separate change.

Reply:
Seems like we could change it here - we're already fixing the logic in minor ways.